pkg/envinfo/url.go

```diff
@@ -62,9 +77,18 @@ func (ei *EnvInfo) CompleteURL(url *localConfigProvider.LocalURL) error {
 	url.Path = "/" + url.Path
 
 	// get the port if not provided
-	ports, err := ei.GetPorts()
-	if err != nil {
-		return err
+	var ports []string
+	var err error
+	if url.Container == "" {
+		ports, err = ei.GetComponentPorts()
+		if err != nil {
+			return err
+		}
+	} else {
+		ports, err = ei.GetContainerPorts(url.Container)
```

Comment:
When no port is found for the specified container, the error message is
```
[mrinaldas@localhost project]$ odo url create example --container runtime-1 --host com
 ✗  no port is exposed by the component nodejs-project-hron, please specify a port
```
It feel it should say `no port is exposed by the container <container-name> ....`

Reply:
Ok pushing some changes